### PR TITLE
core: add documentation on how to get line of failed assert()

### PR DESCRIPTION
**Pull request by miri64**
Requested doc from https://github.com/RIOT-OS/RIOT/pull/5875

### CODE REVIEW

**Comment by miri64** [comment]
Ping?

File: core/include/assert.h [hunk 1]
```diff
@@ -41,6 +41,19 @@ extern const char assert_crash_message[];
  *
  * The purpose of this macro is to help programmers find bugs in their
  * programs.
+ *
+ * The output if an assertion fails and NDEBUG is set is usually something like
```
Comment (miri64) [inline core/include/assert.h]: There are some false assumptions (in the original wording as well):

1. There is a logical error: the output only is provided if `NDEBUG` is not defined
2. There is IMHO a difference between an enabled (or set) macro and a defined macro. To demonstrate lets take the antonyms "disabled" and "undefined": a disabled macro can be defined, but its value is set to 0, while an undefined macro just does not exist.

Except for the wording though I agree with the simpler sentencing. Will fix.

**Comment by miri64** [comment]
Done.

**Comment by x3ro** [comment]
Squash and merge? :)

**Comment by miri64** [comment]
You tell me ;-).

**Comment by miri64** [comment]
Okay to merge?

**Comment by kaspar030** [comment]
Yes!

**Pull request merged**